web/styles/compose.css

```diff
@@ -704,16 +740,16 @@
 
 textarea.new_message_textarea {
     display: table-cell;
-    width: calc(100% - 12px);
     padding: 5px;
     height: 1.5em;
     max-height: 22em;
-    margin-bottom: 0;
+    margin: 0;
     resize: vertical !important;
-    margin-top: 5px;
     border-radius: 4px;
     color: hsl(0deg 0% 33%);
     background-color: hsl(0deg 0% 100%);
+    margin-right: -25px;
+    padding-right: 25px;
```

Comment:
This doesn't feel very grid-like. Is there a reason to do this hackery with padding and a negative margin instead use of named grid-areas to position things where they belong?

Reply:
The design needs these buttons to show up visually inside the textarea, that's why it does need the textarea to be extended using negative margin over to be the backdrop for the buttons.

An alternative could be to have the buttons' grid area be the same background color and same height, with the borders adjusted to look like it's the "same" textarea extended. This is what I have in mind for making the formatting buttons row seem to be "inside" the textarea as well. But here. the main problem is that _the scrollbar will appear weirdly between the textarea instead of being pushed to the right edge_. Hence I went with the negative margin approach here.

Does that make sense @karlstolley?

Reply:
Ah, got it. Looking at this more closely, it seems like the issue is also to keep text in the textarea from running behind the buttons? That's probably okay...I'd just ask that you document that next to the lines above so those values have some adjacent explanation.

Reply:
Yep! Forgot to mention the padding keeps the text from overlapping with the buttons' area. Have added a comment explaining all this

Reply:
Got it. And it occurs to me that, because that 25px value shows up in the grid template, too, it might be worth making a variable of those values so it's a) set in one place, and b) discoverable across the code base.

Reply:
That makes perfect sense! I've defined a variable for this and moved the explanatory comment there too